### PR TITLE
Bug 564 - AvatarView size not available to children.

### LINKED ISSUE #564: [Bug] AvatarView size not available to children.
### Description

When using `AvatarView` and setting the `WidthRequest` and/or the `HeighRequest`, these are not available to the child (such as with `GravatarImageSource`.

### Link to Reproduction Sample

[Repository](https://github.com/GeorgeLeithead/CommunityToolkitMaui/tree/110-GravatarImageSource)

#### XAML example

```csharp
            <mct:AvatarView
                BorderWidth="2"
                CornerRadius="32,0,0,32"
                HeightRequest="128"
                WidthRequest="128">
                <mct:AvatarView.Stroke>
                    <LinearGradientBrush EndPoint="0,1">
                        <GradientStop Offset="0.1" Color="Blue" />
                        <GradientStop Offset="1.0" Color="Red" />
                    </LinearGradientBrush>
                </mct:AvatarView.Stroke>
                <mct:AvatarView.ImageSource>
                    <mct:GravatarImageSource
                        CacheValidity="{Binding CacheValidityTimespan}"
                        CachingEnabled="{Binding EnableCache}"
                        Email="{Binding Email}"
                        Image="{Binding DefaultGravatarSelected}"
                        SemanticProperties.Description="GravatarImageSourse used with AvatarView." />
                </mct:AvatarView.ImageSource>
```

### Steps to Reproduce

1. Add `AvatarView` to page.
2. Set `WidthRequest` and/or `HeightRequest` to a value much larger than 80 (such as 128)
3. Set as the `ImageSource` a valid `GravatarImageSource`.

The image returned by `GravatarImageSource` will be blurry as it can't get the parent `AvatarView` size and thus will size to the default of 80x80.

### Expected Behavior
The `ImageSource` size should be able to be retrieved from the parent.

E.g. If the `HeightRequest` was 128: https://www.gravatar.com/avatar/b65a519785f69fbe7236dd0fd6396094?s=128&d=mp

### Actual Behavior
The `ImageSource` size returned by `GravatarImageSource` defaults to 80x80, if the parent Height\request and/or Width\request cant not be retrieved.

E.g. https://www.gravatar.com/avatar/b65a519785f69fbe7236dd0fd6396094?s=80&d=mp

### Basic Information

- Version with issue: 1.2.0
- Last known good version: N/A
- IDE:
- Platform Target Frameworks: 
  - iOS:  ALL
  - Android: ALL
  - UWP:  ALL
- Android Support Library Version: ALL
- Nuget Packages: ALL
- Affected Devices: ALL

### Reproduction imagery
image

### PR DESCRIPTION
**Pull request by GeorgeLeithead**
### Description of Change ###

Bind `WidthRequest` and `HeightRequest` from parent to child `Image`.

#### Additional ####
Samples has been updated to include using `UriImageSource` for an image avatar.
NOTE: That the origin of this bug was identified with #531 and the samples in this branch better reflect the issue and the fix.

### Linked Issues ###

 - Fixes #564 

### PR Checklist ###

 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests
 - [x] Has samples
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [x] Documentation created or updated: N/A 

#### Platforms tested ####
- [x] Android
- [x] Windows
- [ ] iOS
- [ ] MacCatalyst

### CODE REVIEW

**Comment by GeorgeLeithead** [comment]
Additional unit tests are needed for completness!

File: src/CommunityToolkit.Maui/Views/AvatarView.shared.cs [hunk 1]
```diff
@@ -72,6 +72,8 @@ public AvatarView()
 			CornerRadius = new CornerRadius(AvatarViewDefaults.DefaultCornerRadius.TopLeft, AvatarViewDefaults.DefaultCornerRadius.TopRight, AvatarViewDefaults.DefaultCornerRadius.BottomLeft, AvatarViewDefaults.DefaultCornerRadius.BottomRight),
 		};
 		Content = avatarLabel;
+		avatarImage.SetBinding(WidthRequestProperty, new Binding(nameof(WidthRequest), source: this));
+		avatarImage.SetBinding(HeightRequestProperty, new Binding(nameof(HeightRequest), source: this));
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Views/AvatarView.shared.cs:76]: These need to be bound to `Parent.WidthRequest` and `Parent.HeightRequest`. Currently, they are just binding to themselves via `source: this`.

```suggestion
		avatarImage.SetBinding(WidthRequestProperty, new Binding(nameof(WidthRequest), source: Parent));
		avatarImage.SetBinding(HeightRequestProperty, new Binding(nameof(HeightRequest), source: Parent));
```

File: src/CommunityToolkit.Maui/Views/AvatarView.shared.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@
 namespace CommunityToolkit.Maui.Views;
 
 /// <summary>AvatarView control.</summary>
-public class AvatarView : Border, IAvatarView, IBorderElement, IFontElement, ITextElement, IImageElement, ITextAlignmentElement, ILineHeightElement, ICornerElement, IPaddingElement
```
Comment (brminnick) [inline src/CommunityToolkit.Maui/Views/AvatarView.shared.cs:9]: How come `IPaddingElement` is being removed?

**Comment by GeorgeLeithead** [comment]
There is more than just the fix for making the size available to child `image`, but also a fix for `ImageSource` binding, removing un-necessary inheritance, and naming consolidation.  Also, the unit tests have gone from 62.8% coverage to 99.3%.

**Comment by GeorgeLeithead** [comment]
@brminnick @bijington I believe that this PR is in a good state to be added to a release, **just needing review**.  There is more than just the fix for making the size available to child image, but also a fix for ImageSource binding, removing un-necessary inheritance, and naming consolidation. Also, the unit tests have gone from 62.8% coverage to 99.3%.

**Review by brminnick — APPROVED** [review]
Thanks George!

**Pull request merged**